### PR TITLE
Add `io-function-sign-dss` to `io-sign`

### PR DESCRIPTION
**Pull request by grausof**
### List of changes

1. Add new **JAVA** azure function in `io-sign`  

### Type of changes

- [X] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [X] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [X] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

**Comment by grausof** [comment]
this PR depends on #235

File: src/domains/sign/io_sign_dss_func.tf [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  linux_fx_version  = "JAVA|11-java11"
+  health_check_path = "api/info"
+
+
+  app_settings = {
+    FUNCTIONS_WORKER_RUNTIME     = "java"
+    WEBSITE_VNET_ROUTE_ALL       = "1"
+    WEBSITE_DNS_SERVER           = "168.63.129.16"
+    QueueStorageConnection       = data.azurerm_storage_account.api.primary_connection_string
+    CosmosDbConnectionString     = data.azurerm_key_vault_secret.cosmosdb_connection_string.value
```
Comment (pasqualedevita) [inline src/domains/sign/io_sign_dss_func.tf:26]: Better use reference to resources

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

File: src/domains/sign/io_sign_dss_func.tf [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    IOApiSubscriptionKey         = data.azurerm_key_vault_secret.io_api_subscription_key.value
+    StorageAccountConnectionString = data.azurerm_key_vault_secret.storage_account_connection_string.value
+  }
+
+  allowed_subnets = [module.io_sign_snet.id, data.azurerm_subnet.apim.id]
```
[the remark this answers is not in the document]
Comment (grausof) [reply src/domains/sign/io_sign_dss_func.tf:31]: nobody. These are all functions triggered by queue!

File: src/domains/sign/io_sign_dss_func.tf (hunk 1 above)
Comment (grausof) [reply src/domains/sign/io_sign_dss_func.tf:26]: Thanks!

File: src/domains/sign/io_sign_dss_func.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (pasqualedevita) [reply src/domains/sign/io_sign_dss_func.tf:31]: perfect, so we can remove 
```suggestion
```

**Pull request closed without merging**